code/entities/cooking/hotdog-roller/components/interactions/switches/BaseSwitch.cs

```diff
@@ -0,0 +1,43 @@
+﻿using Sandbox;
+
+namespace Cinema;
+
+public partial class BaseSwitch : EntityComponent<HotdogRoller>
+{
+    /// <summary>
+    /// Sets switch state
+    /// </summary>
+    public void SetPos(int pos)
+    {
+        State state = (State)pos;
+
+        TransitionStateTo(state);
+    }
+
+    /// <summary>
+    /// Toggles state between State On and State Off
+    /// </summary>
+    public void TogglePos()
+    {
+        if(SwitchState == State.Off)
```

Comment:
If you make SwitchState a boolean then you can do `TransitionStateTo(!isRolling)`